From 8082956dff27577b4eed6a5daab2ecea4dfd4adf Mon Sep 17 00:00:00 2001
From: huangxin <huangxin@MServer.allwinnertech.com>
Date: Mon, 28 Nov 2011 10:23:56 +0800
Subject: [PATCH 387/944] audiocodec:del the mdelay time when play and capture
 open

---
 sound/soc/sun4i/sun4i-codec.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/sound/soc/sun4i/sun4i-codec.c b/sound/soc/sun4i/sun4i-codec.c
index 00f6f1d..e04c5ad 100644
--- a/sound/soc/sun4i/sun4i-codec.c
+++ b/sound/soc/sun4i/sun4i-codec.c
@@ -360,7 +360,6 @@ static  int codec_init(void)
 static int codec_play_open(struct snd_pcm_substream *substream)
 {
 	codec_wr_control(SUN4I_DAC_DPC ,  0x1, DAC_EN, 0x1);
-	mdelay(100);
 	codec_wr_control(SUN4I_DAC_FIFOC ,0x1, DAC_FIFO_FLUSH, 0x1);
 	//set TX FIFO send drq level
 	codec_wr_control(SUN4I_DAC_FIFOC ,0x4, TX_TRI_LEVEL, 0xf);
@@ -400,7 +399,6 @@ static int codec_capture_open(void)
 	 codec_wr_control(SUN4I_ADC_FIFOC, 0xf, RX_TRI_LEVEL, 0x7);
 	 //enable adc1 analog
 	 codec_wr_control(SUN4I_ADC_ACTL, 0x3,  ADC_EN, 0x3);
-	 mdelay(100);
 	 return 0;
 }
 
-- 
1.8.0

