From fc292415ea1b668a80f77454e6d97df2e95f2da3 Mon Sep 17 00:00:00 2001
From: Tom Cubie <tangliang@allwinnertech.com>
Date: Wed, 30 Nov 2011 16:24:14 +0800
Subject: [PATCH 401/944] pack: make boot.img default

---
 scripts/build_sun4i.sh | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/scripts/build_sun4i.sh b/scripts/build_sun4i.sh
index 27bce3f..7e036f1 100755
--- a/scripts/build_sun4i.sh
+++ b/scripts/build_sun4i.sh
@@ -76,6 +76,12 @@ build_kernel()
 	cp .config output/
 	cp rootfs/sun4i_rootfs.cpio.gz output/
 
+	mkbootimg --kernel output/bImage \
+			--ramdisk output/sun4i_rootfs.cpio.gz \
+			--board 'sun4i' \
+			--base 0x40000000 \
+			-o output/boot.img
+
 
 	for file in $(find drivers sound crypto block fs security net -name "*.ko"); do
 		cp $file ${LICHEE_MOD_DIR}
@@ -176,7 +182,7 @@ clean_modules()
 #####################################################################
 
 LICHEE_ROOT=`(cd ${LICHEE_KDIR}/..; pwd)`
-export PATH=${LICHEE_ROOT}/buildroot/output/external-toolchain/bin:$PATH
+export PATH=${LICHEE_ROOT}/buildroot/output/external-toolchain/bin:${LICHEE_ROOT}/tools/pack/pctools/linux/android:$PATH
 
 case "$1" in
 kernel)
-- 
1.8.0

