From 11b4e18376d662a57d3c0296bd8ea571cba33327 Mon Sep 17 00:00:00 2001
From: Alejandro Mery <amery@geeks.cl>
Date: Sun, 4 Mar 2012 15:28:45 +0100
Subject: [PATCH 437/944] ata: sun4i: adapted for 3.0

---
 drivers/ata/sw_ahci_platform.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/ata/sw_ahci_platform.c b/drivers/ata/sw_ahci_platform.c
index de4cc75..6e3c6c8 100644
--- a/drivers/ata/sw_ahci_platform.c
+++ b/drivers/ata/sw_ahci_platform.c
@@ -375,9 +375,6 @@ static int __init sw_ahci_probe(struct platform_device *pdev)
 		ata_port_desc(ap, "mmio %pR", mem);
 		ata_port_desc(ap, "port 0x%x", 0x100 + ap->port_no * 0x80);
 
-		/* set initial link pm policy */
-		ap->pm_policy = NOT_AVAILABLE;
-
 		/* set enclosure management message type */
 		if (ap->flags & ATA_FLAG_EM)
 			ap->em_message_type = hpriv->em_msg_type;
-- 
1.8.0

