From 732cff8519b040e5750a7d468570e3a2a43b65d1 Mon Sep 17 00:00:00 2001
From: Alejandro Mery <amery@geeks.cl>
Date: Fri, 1 Jun 2012 22:02:28 +0200
Subject: [PATCH 495/944] arm: head.S: move mangled r2 to 0x81000000

---
 arch/arm/kernel/head.S | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/kernel/head.S b/arch/arm/kernel/head.S
index 6b81595..dc6a391 100644
--- a/arch/arm/kernel/head.S
+++ b/arch/arm/kernel/head.S
@@ -98,15 +98,15 @@ ENTRY(stext)
 #if CONFIG_CHIP_ID == 1120
 #warning "chip id: 1120"
 1:	.long 0xf37
-	.long 0x40000000
+	.long 0x81000000
 #elif CONFIG_CHIP_ID == 1123
 #warning "chip id: 1123"
 1:	.long 0xf34
-	.long 0x40000000
+	.long 0x81000000
 #elif CONFIG_CHIP_ID == 1125
 #warning "chip id: 1125"
 1:      .long 0xf35
-        .long 0x40000000
+        .long 0x81000000
 #else
 #error "unsupport chip"
 #endif
-- 
1.8.0

