From 7f5ba1bc192c8f6750f191f306194d8aa87a04fc Mon Sep 17 00:00:00 2001
From: Alejandro Mery <amery@geeks.cl>
Date: Mon, 16 Jul 2012 14:03:35 +0200
Subject: [PATCH 615/944] ARM: sun4i: nicer printing for reserved memory areas

---
 arch/arm/mach-sun4i/core.c | 22 +++++++++++++---------
 1 file changed, 13 insertions(+), 9 deletions(-)

diff --git a/arch/arm/mach-sun4i/core.c b/arch/arm/mach-sun4i/core.c
index 99f1ceb..136fbe2 100644
--- a/arch/arm/mach-sun4i/core.c
+++ b/arch/arm/mach-sun4i/core.c
@@ -148,6 +148,12 @@ static void __init sw_core_fixup(struct tag *t, char **cmdline,
 		pr_info("%u MB reserved for MALI\n", mali);
 }
 
+#define pr_reserve_info(L, START, SIZE) \
+	pr_info("\t" L " : 0x%08x - 0x%08x  (%4d %s)\n", \
+		(u32)(START), (u32)((START) + (SIZE) - 1), \
+		(u32)((SIZE) < SZ_1M ? (SIZE) / SZ_1K : (SIZE) / SZ_1M), \
+		(SIZE) < SZ_1M ? "kB" : "MB")
+
 /* Only reserve certain important memory blocks if there are actually
  * drivers which use them.
  */
@@ -176,7 +182,7 @@ static void __init reserve_fb(void)
     if (sw_cfg_get_int(script_base, "disp_init", "disp_init_enable"))
     {
 		memblock_reserve(fb_start, fb_size);
-		pr_info("\tLCD: 0x%08x, 0x%08x\n", (unsigned int)fb_start, (unsigned int)fb_size);
+		pr_reserve_info("LCD ", fb_start, fb_size);
     }
 	else
 		fb_start = fb_size = 0;
@@ -214,7 +220,7 @@ static void __init reserve_g2d(void)
 		g2d_size = g2d_size;
 		memblock_reserve(g2d_start, g2d_size);
 
-		pr_info("\tG2D: 0x%08x, 0x%08x\n", (unsigned int)g2d_start, (unsigned int)g2d_size);
+		pr_reserve_info("G2D ", g2d_start, g2d_size);
     }
     else
     	g2d_start = g2d_size = 0;
@@ -248,7 +254,7 @@ static void __init reserve_ve(void)
 	memblock_reserve(ve_start, SZ_64M);
 	memblock_reserve(ve_start + SZ_64M, SZ_16M);
 
-	pr_info("\tVE : 0x%08x, 0x%08x\n", (unsigned int)ve_start, (unsigned int)ve_size);
+	pr_reserve_info("VE  ", ve_start, ve_size);
 }
 
 #else
@@ -258,18 +264,16 @@ static void __init reserve_ve(void) {}
 static void reserve_sys(void)
 {
 	memblock_reserve(SYS_CONFIG_MEMBASE, SYS_CONFIG_MEMSIZE);
-	pr_info("\tSYS: 0x%08x, 0x%08x\n",
-			(unsigned int)SYS_CONFIG_MEMBASE,
-			(unsigned int)SYS_CONFIG_MEMSIZE);
+	pr_reserve_info("SYS ", SYS_CONFIG_MEMBASE, SYS_CONFIG_MEMSIZE);
 }
 
 static void __init sw_core_reserve(void)
 {
-	pr_info("Memory Reserved(in bytes):\n");
+	pr_info("Memory Reserved:\n");
 	reserve_sys();
-	reserve_fb();
-	reserve_g2d();
 	reserve_ve();
+	reserve_g2d();
+	reserve_fb();
 }
 
 void sw_irq_ack(struct irq_data *irqd)
-- 
1.8.0

