From eda626048b3fcc3f884a62942da336dc5a6e8ab0 Mon Sep 17 00:00:00 2001
From: Henrik Nordstrom <henrik@henriknordstrom.net>
Date: Thu, 19 Jul 2012 00:14:32 +0200
Subject: [PATCH 616/944] sunxi: Correct framebuffer & lcd/hdmi Kconfig
 dependencies

---
 drivers/video/Kconfig | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/video/Kconfig b/drivers/video/Kconfig
index 1dfa26c..181859a 100644
--- a/drivers/video/Kconfig
+++ b/drivers/video/Kconfig
@@ -280,7 +280,7 @@ config SUN3I_FB
 
 config SUN3I_HDMI
 	tristate "HDMI Driver Support(sun3i)"
-	depends on ARCH_SUN3I
+	depends on SUN3I_FB
 	---help---
 	default m
 
@@ -292,18 +292,18 @@ config SUN4I_FB
 
 config SUN4I_FB_UMP
 	bool "Enable FB/UMP Integration"
-	depends on MALI=m && SUN4I_FB=m
+	depends on MALI && SUN4I_FB=m
 	default y
 
 config SUN4I_LCD
         tristate "LCD Driver Support(sun4i)"
-        depends on ARCH_SUN4I
+        depends on SUN4I_FB
         ---help---
         default m
 
 config SUN4I_HDMI
         tristate "HDMI Driver Support(sun4i)"
-        depends on ARCH_SUN4I
+        depends on SUN4I_FB
         ---help---
         default m
 
@@ -315,13 +315,13 @@ config SUN5I_FB
 
 config SUN5I_LCD
         tristate "LCD Driver Support(sun5i)"
-        depends on ARCH_SUN5I
+        depends on SUN5I_FB
         ---help---
         default m
 
 config SUN5I_HDMI
         tristate "HDMI Driver Support(sun5i)"
-        depends on ARCH_SUN5I
+        depends on SUN5I_FB
         ---help---
         default m
 
-- 
1.8.0

